src/main/angular/model/types.ts

```diff
@@ -0,0 +1,22 @@
+/** Contains various types used by the application. */
+
+/** Properties used by GCharts for rendering. */
+export type GraphProperties = {
+  options: google.visualization.LineChartOptions;
+  graphData: Row[];
+  columns: Columns;
+  type: 'LineChart';
+  title: string;
+  width: number;
+  height: number;
+};
+
+/** The different types of data permitted in a row for Google Charts. */
+// Timestamp is a date, number bindings is a number, tooltip is a string, styling is a string.
+export type RowData = Date | string | number | undefined;
```

Comment:
undefined... hmm... in what situations is the RowData null?

Reply:
I added a comment. It's when the series lining up with those columns is not present at that point in time.